uint16 val_x1
uint16 val_y1
uint16 val_x2
uint16 val_y2
